modules/apps/portal-template/portal-template-freemarker/src/main/java/com/liferay/portal/template/freemarker/internal/helper/FreeMarkerTemplateContextHelper.java

```diff
@@ -154,12 +152,23 @@ public void setRestrictedBeansWrapper(BeansWrapper restrictedBeansWrapper) {
 
 	@Activate
 	@Modified
-	protected void activate(Map<String, Object> properties) {
+	protected void activate(
+		BundleContext bundleContext, Map<String, Object> properties) {
+
 		_freeMarkerEngineConfiguration = ConfigurableUtil.createConfigurable(
 			FreeMarkerEngineConfiguration.class, properties);
 
 		_restrictedVariables = SetUtil.fromArray(
 			_freeMarkerEngineConfiguration.restrictedVariables());
+
+		_serviceTrackerList = ServiceTrackerListFactory.open(
```

Comment:
@W2jx Noticed a new problem recently.

See https://github.com/liferay-core-infra/liferay-portal/pull/2485#pullrequestreview-1189223037 for my explanation.

In short, this `activate` method is also annotated with `@Modified`, so when the configuration of this component changes, the `activate` method will be called again without calling `deactivate`. This means the previous service tracker list will not be closed when you create a new one.

Please move the configuration related logic to a private method, and add a new method to take the `@Modified` annotation. Then you can call the private method from both `activate()` and `modified()`.